### PR TITLE
docs: allow to set doc_struct_name for global functions

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
Some of those are moved to a different struct in some cases
like gdk::Key and would be nice to have them documented

### CODE REVIEW

File: src/codegen/doc/mod.rs [hunk 1]
```diff
@@ -179,7 +179,18 @@ fn generate_doc(w: &mut dyn Write, env: &Env) -> Result<()> {
...
                     .flatten();
-                let parent = doc_trait_name.map(|p| Box::new(TypeStruct::new(SType::Trait, p)));
+                let doc_struct_name = (&global_functions.functions)
+                    .iter()
+                    .find(move |f| &f.glib_name == c_identifier)
+                    .map(|f| f.doc_struct_name.as_ref())
+                    .flatten();
+                let parent = if let Some(doc_trait_name) = doc_trait_name {
+                    Some(Box::new(TypeStruct::new(SType::Trait, doc_trait_name)))
+                } else if let Some(doc_struct_name) = doc_struct_name {
```
Comment (sdroege) [inline src/codegen/doc/mod.rs:189]: clippy is angry

**Comment by sdroege** [comment]
Seems good to me otherwise

File: src/codegen/doc/mod.rs (hunk 1 above)
Comment (bilelmoussaoui) [reply src/codegen/doc/mod.rs]: Fixed

File: src/codegen/doc/mod.rs [hunk 2]
```diff
@@ -179,7 +179,24 @@ fn generate_doc(w: &mut dyn Write, env: &Env) -> Result<()> {
                     .find(move |f| &f.glib_name == c_identifier)
                     .map(|f| f.doc_trait_name.as_ref())
                     .flatten();
-                let parent = doc_trait_name.map(|p| Box::new(TypeStruct::new(SType::Trait, p)));
+                let doc_struct_name = (&global_functions.functions)
+                    .iter()
+                    .find(move |f| &f.glib_name == c_identifier)
+                    .map(|f| f.doc_struct_name.as_ref())
+                    .flatten();
```
Comment (sdroege) [inline src/codegen/doc/mod.rs:186]:
```suggestion
                    .and_then(|f| f.doc_struct_name.as_ref());
```

?

Comment (bilelmoussaoui) [reply src/codegen/doc/mod.rs]: ah, for whatever reason i didn't notice we have the same thing a few lines above that uses and_then :)

**Review by sdroege — APPROVED** [review]

**Comment by bilelmoussaoui** [comment]
I pushed another commit to not generate docs for functions that are not generated, this reduces the size of the generated docs.txt (to hopefully completely remove it which means all the generated docs gets embedded properly)

**Comment by bilelmoussaoui** [comment]
Don't merge yet, i noticed part of 9bb9c111cb945130ac2814b2996377f9d427f7fc that doesn't behave as it should

**Comment by bilelmoussaoui** [comment]
Should be good to go now, there are more stuff to cleanup but are unrelated to those changes

**Review by sdroege — APPROVED** [review]
@GuillaumeGomez ?

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**